### DISCUSSION TITLE
Custom Asset Manager - Update selected with new asset

### DISCUSSION THREAD

**Discussion by JamesHusband** [GrapesJS/grapesjs#4687] [Q&A]
I have my custom asset manager working, when a new asset is uploaded it is added to the **Assets** array. My question is what function should I call to update the selected asset (in the editor) which new asset uploaded via Asset Manager?

**Comment by artf** [comment]
Hi @JamesHusband the update logic should be implemented by the caller of the asset manager https://grapesjs.com/docs/modules/Assets.html#custom-select-logic as it might be different for each component.
Here you can see how the built-in image component uses the Asset Manager
https://github.com/artf/grapesjs/blob/01352d366dbbd846e8dd0d29249062814cf79384/src/dom_components/view/ComponentImageView.js#L61-L77
